pkg/virt-operator/resource/generate/components/crds.go

```diff
@@ -705,6 +711,52 @@ func NewMigrationPolicyCrd() (*extv1.CustomResourceDefinition, error) {
 	return crd, nil
 }
 
+func NewVirtualMachineCloneCrd() (*extv1.CustomResourceDefinition, error) {
+	crd := newBlankCrd()
+
+	crd.ObjectMeta.Name = VIRTUALMACHINECLONE
+	crd.Spec = extv1.CustomResourceDefinitionSpec{
+		Group: clonev1alpha1.VirtualMachineCloneKind.Group,
+		Versions: []extv1.CustomResourceDefinitionVersion{
+			{
+				Name:    clonev1alpha1.SchemeGroupVersion.Version,
+				Served:  true,
+				Storage: true,
+			},
+		},
+		Scope: extv1.NamespaceScoped,
+
+		Names: extv1.CustomResourceDefinitionNames{
+			Plural:     clone.ResourceVMClonePlural,
+			Singular:   clone.ResourceVMCloneSingular,
+			ShortNames: []string{"vmclone", "vmclones"},
+			Kind:       clonev1alpha1.VirtualMachineCloneKind.Kind,
+			Categories: []string{
+				"all",
+			},
+		},
+	}
+	err := addFieldsToAllVersions(crd,
+		&extv1.CustomResourceSubresources{
+			Status: &extv1.CustomResourceSubresourceStatus{},
+		},
+		[]extv1.CustomResourceColumnDefinition{
+			{Name: "Age", Type: "date", JSONPath: creationTimestampJSONPath},
+			{Name: "Phase", Type: "string", JSONPath: phaseJSONPath},
+			{Name: "SnapshotName", Type: "string", JSONPath: ".status.snapshotName"},
+			{Name: "RestoreName", Type: "string", JSONPath: ".status.restoreName"},
```

Comment:
Since SnapshotName/RestoreName are transitory now (deleted when vm created) I think it is confusing to leave them here

Also, not sure about Age field.

Reply:
Sure.
I've removed these. I've also added `SourceVirtualMachine` and `TargetVirtualMachine`, thought it would be valuable (not enforce people to look at the spec). It's also valuable for clones that do not explicitly fill the "target" field, which means that the target would have an auto-generated name.